indico/modules/users/controllers.py

```diff
@@ -177,13 +183,63 @@ class RHPersonalData(RHUserBase):
     def _process(self):
         form = UserDetailsForm(obj=FormDefaults(self.user, skip_attrs={'title'}, title=self.user._title),
                                synced_fields=self.user.synced_fields, synced_values=self.user.synced_values)
+        pic_form = UserPictureForm()
         if form.validate_on_submit():
             self.user.synced_fields = form.synced_fields
             form.populate_obj(self.user, skip=self.user.synced_fields)
             self.user.synchronize_data(refresh=True)
             flash(_('Your personal data was successfully updated.'), 'success')
             return redirect(url_for('.user_profile'))
-        return WPUser.render_template('personal_data.html', 'personal_data', user=self.user, form=form)
+        elif self.user.has_picture:
+            pic_form.picture.data = self.user
+        return WPUser.render_template('personal_data.html', 'personal_data', user=self.user, form=form,
+                                      pic_form=pic_form)
+
+
+class RHProfilePictureDisplay(RHUserBase):
+    def _process(self):
+        if not self.user.has_picture:
+            raise NotFound
+        metadata = self.user.picture_metadata
+        return send_file('user.png', BytesIO(self.user.picture), mimetype=metadata['content_type'],
+                         conditional=True)
+
+
+class RHProfilePicture(RHUserBase):
+    def _process(self):
+        f = request.files['picture']
+        try:
+            pic = Image.open(f)
+        except IOError:
+            raise UserValueError(_('You cannot upload this file as profile picture.'))
+        if pic.format.lower() not in {'jpeg', 'png', 'gif'}:
+            raise UserValueError(_('The file has an invalid format ({format}).').format(format=pic.format))
+        if pic.mode not in {'RGB', 'RGBA'}:
+            pic = pic.convert('RGB')
+        image_bytes = BytesIO()
+        pic = square(pic)
```

Comment:
I think we also need to resize the image to a max size (256x256px or something like that).

I just managed to upload an 8mb photo of a very pretty cat, but as much as I like hi-res cat pictures, a user picture/avatar is not the place for them ;)